cmd/clusternetwork/config.go

```diff
@@ -0,0 +1,70 @@
+package clusternetwork
+
+import (
+	configclient "github.com/openshift/client-go/config/clientset/versioned"
+	operatorv1 "github.com/openshift/client-go/operator/clientset/versioned/typed/operator/v1"
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ClusterNetworkConfig interface contains methods to validate network configuration of a cluster
+type ClusterNetworkConfig interface {
+	Validate() error
+}
+
+// OVNKubernetes contains information specific to network type OVNKubernetes
+type OVNKubernetes struct {
+	// name describes value of the Network Type
+	name string
+	// clienset is the OpenShift config client, we will use to query the OpenShift api object status
+	clienset configclient.Interface
+	// operatorClient is the OpenShift operator client, we will use to interact with OpenShift operator objects
+	operatorClient operatorv1.OperatorV1Interface
+}
+
+// validate for Ovn kubernetes checks for network type and hybrid overlay.
+func (ov *OVNKubernetes) Validate() error {
+	//check if hybrid overlay is enabled for the cluster
+	networkCR, err := ov.operatorClient.Networks().Get("cluster", metav1.GetOptions{})
+	if err != nil {
+		return errors.Wrap(err, "error getting cluster network object")
+	}
+
+	defaultNetwork := networkCR.Spec.DefaultNetwork
+	if defaultNetwork.OVNKubernetesConfig == nil || defaultNetwork.OVNKubernetesConfig.HybridOverlayConfig == nil {
+		return errors.New("hybrid overlay is not enabled in the cluster")
+	}
+
+	if len(networkCR.Spec.DefaultNetwork.OVNKubernetesConfig.HybridOverlayConfig.HybridClusterNetwork) == 0 {
+		return errors.New("no hybrid cluster network found")
+	}
+	return nil
+}
+
+// networkConfigurationFactory is a factory method that returns information specific to network type
+func NetworkConfigurationFactory(clientset configclient.Interface, operatorClient operatorv1.OperatorV1Interface) (ClusterNetworkConfig, error) {
+	networkType, err := getNetworkType(clientset)
+	if err != nil {
+		return nil, err
+	}
+	switch networkType {
+	case "OVNKubernetes":
+		return &OVNKubernetes{
+			name:           networkType,
+			clienset:       clientset,
+			operatorClient: operatorClient,
+		}, nil
+	default:
+		return nil, errors.Errorf("%s : network type not supported", networkType)
+	}
+}
+
+// getNetworkType returns network type of the cluster
+func getNetworkType(clientset configclient.Interface) (string, error) {
+	// Get the cluster network object so that we can find the network type
+	networkCR, err := clientset.ConfigV1().Networks().Get("cluster", metav1.GetOptions{})
```

Comment:
Should we update our RBAC to get network object?

Reply:
good catch. this is necessary